### DISCUSSION TITLE
[Feature Request] Add option to toggle between gpu modes

### DISCUSSION THREAD

**Discussion by bxfy1** [seerge/g-helper#1510] [Ideas]
Hi, currently for key binding you can bind it to "**Mode**" for switching between different cpu profiles.

Can the same ability be added for switching between gpu modes?

**Comment by seerge** [comment]
@bxfy1 hello,

GPU "modes" are actually only to turn dGPU on or off :) it's quite invasive action, and it's better when user does that consciously via UI 

If you want it automatically disable dGPU on battery, and turn in back on when plugged in - just select Optimized

**Comment by bxfy1** [comment]
The actual intent was to disable the dGPU when plugged in to lower device's temperature and fan noise while performing low load tasks.

Anyways, not a problem if the feature is not desired.

I've created a simple one click script to achieve the toggle effect as a work around. Saved a macro key in the process :P

**Reply by RajeshKumarRaja** [reply]
Can you share the script please? I dont know where to start for the macro.

**Reply by seerge** [reply]
@RajeshKumarRaja you can toggle GPU modes with hotkey-combos now. You don't need any scripts for that
https://github.com/seerge/g-helper?tab=readme-ov-file#%EF%B8%8F-keybindings

**Reply by RajeshKumarRaja** [reply]
I want to switch to ECO mode and the hotkey combination is Ctrl + Shift + Alt + F14. Whereas my asus tuf F15 don't have this key F14. Plus I want the iGPU mode to automatically switch to ECO mode on start-up to save energy.

**Reply by seerge** [reply]
You can always use Sharpkeys or Autohotkey to bind your preferred key to emit  Ctrl + Shift + Alt + F14. 

For automatic dGPU disabling just use Optimized mode. It does that already from the beginning.
